docs/src/jekyll/docs.md

```diff
@@ -0,0 +1,546 @@
+
+# Usage
+
+In order to tell Fetch how to retrieve data, we must implement the `DataSource` typeclass.
+
+```scala
+trait DataSource[Identity, Result]{
+  def fetch(ids: List[Identity]): Eval[Map[Identity, Result]]
+}
+```
+
+It takes two type parameters:
+
+ - `Identity`: the identity we want to fetch (a `UserId` if we were fetching users)
+ - `Result`: the type of the data we retrieve (a `User` if we were fetching users)
+
+The `fetch` method takes a non empty list of identities and must return an `Eval` that will result
+in a map from identities to results. Accepting a list of identities gives Fetch the ability to batch requests to
+the same data source, and returning a mapping from identities to results Fetch can detect whenever an identity
+couldn't be fetched or no longer exists.
+
+Returning `Eval` makes it possible to defer evaluation with a concurrency monad when running a fetch. For example
+when using `Future` as your target monad when running a fetch the data source fetches will possibly run in a thread pool,
+depending on the implicit execution context you use.
+
+## Writing your first data source
+
+Now that we know about the `DataSource` typeclass, let's write our first data source! We'll be using Scala's
+`Future` as our concurrency monad for the examples. We'll write a little function for simulating latency and
+another for blocking on futures to get their results. Note that the `Future` API differs in Scala and Scala.js,
+since there are no threads in JavaScript and you cannot block for a `Future`'s result.
+
+```scala
+import scala.util.Random
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+import cats.Eval
+
+def latency[A](result: A, msg: String, wait: Int = Random.nextInt(100)): Eval[A] =
+  Eval.later({
+    val threadId = Thread.currentThread().getId()
+    println(s"~~~>[thread: $threadId] $msg")
+    Thread.sleep(wait)
+    println(s"<~~~[thread: $threadId] $msg")
+    result
+  })
+
+def await[A](f: Future[A]): A = Await.result(f, 5 seconds)
+```
+
+We'll start implementing a data source for fetching users given their id. The identity type will be a user's
+id and the result type a user.
+
+```scala
+import fetch._
+
+type UserId = Int
+case class User(id: UserId, username: String)
+
+implicit object UserSource extends DataSource[UserId, User]{
+  override def fetch(ids: List[UserId]): Eval[Map[UserId, User]] = {
+    val result = ids.map(i => (i, User(i, s"@egg_$i"))).toMap
+    latency(result, s"Fetching users $ids")
+  }
+}
+```
+
+Now that we have a data source we can write a function for fetching users
+given an id, we just have to pass a `UserId` as an argument to `Fetch`.
+
+```scala
+def getUser(id: UserId): Fetch[User] = Fetch(id) // or, more explicitly: Fetch(id)(UserSource)
+```
+
+## Declaring and running a fetch
+
+We are now ready to declare and run fetches, the only thing missing is an instance
```

Comment:
This is nice but probably belongs in an advance section of the docs when you want to create your custom runtimes. Since we are gonna provide implicits for most runtimes Task, Future, etc... this is only a special case for those implementing their own.